db/migrate.go

```diff
@@ -0,0 +1,50 @@
+package db
+
+import (
+	"fmt"
+	"io/fs"
+	"net/http"
+
+	"github.com/golang-migrate/migrate/v4"
+	_ "github.com/golang-migrate/migrate/v4/database"
+	_ "github.com/golang-migrate/migrate/v4/database/mysql"
+	_ "github.com/golang-migrate/migrate/v4/database/postgres"
+	_ "github.com/golang-migrate/migrate/v4/source/file"
+	"github.com/golang-migrate/migrate/v4/source/httpfs"
+)
+
+func RunMigrations(config Config, embeddedMigrations fs.FS, resourcePath string) error {
+	m, err := getMigrationInstance(config, embeddedMigrations, resourcePath)
+	if err != nil {
+		return err
+	}
+
+	err = m.Up()
+	if err == migrate.ErrNoChange || err == nil {
+		return nil
+	}
+
+	return err
+}
+
+func RunRollback(config Config, embeddedMigrations fs.FS, resourcePath string) error {
+	m, err := getMigrationInstance(config, embeddedMigrations, resourcePath)
+	if err != nil {
+		return err
+	}
+
+	err = m.Steps(-1)
+	if err == migrate.ErrNoChange || err == nil {
+		return nil
+	}
+
+	return err
+}
+
+func getMigrationInstance(config Config, embeddedMigrations fs.FS, resourcePath string) (*migrate.Migrate, error) {
+	src, err := httpfs.New(http.FS(embeddedMigrations), resourcePath)
```

Comment:
@spy16 @mabdh Any suggestion on which driver we should use? I see in some projects we are using `httpfs` and in some `iofs`

Reply:
`httpfs` afaik more towards serving web pages? so probably iofs is better for this use cases. cmiiw @spy16

Reply:
We can directly use `iofs` so we don't have to wrap the fs in `http.FileSystem`.

Reply:
Fixed